backend_new/src/services/user.service.ts

```diff
@@ -215,21 +241,519 @@ export class UserService {
+          .map((juris) => ({ id: juris.id }))
+          .concat(dto.jurisdictions);
+
+        const listings = existingUser.listings
+          .map((juris) => ({ id: juris.id }))
+          .concat(dto.listings);
+
+        const res = this.prisma.userAccounts.update({
+          include: view,
+          data: {
+            jurisdictions: {
+              connect: jursidictions.map((juris) => ({ id: juris.id })),
+            },
+            listings: {
+              connect: listings.map((listing) => ({ id: listing.id })),
+            },
+            userRoles: {
+              create: {
+                ...dto.userRoles,
+              },
+            },
+          },
+          where: {
+            id: existingUser.id,
+          },
+        });
+
+        return mapTo(User, res);
+      } else {
+        // existing user && ((partner user -> trying to recreate user) || (public user -> trying to recreate a public user))
+        throw new BadRequestException('emailInUse');
+      }
+    }
+
+    let passwordHash = '';
+    if (forPartners) {
+      passwordHash = await passwordToHash(
+        crypto.randomBytes(8).toString('hex'),
+      );
+    } else if (dto instanceof UserCreate) {
+      passwordHash = await passwordToHash(dto.password);
+    }
+
+    let newUser = await this.prisma.userAccounts.create({
+      data: {
+        passwordHash: passwordHash,
+        email: dto.email,
+        firstName: dto.firstName,
+        middleName: dto.middleName,
+        lastName: dto.lastName,
+        dob: dto.dob,
+        phoneNumber: dto.phoneNumber,
+        language: dto.language,
+        mfaEnabled: forPartners,
+        jurisdictions: {
+          connect: dto.jurisdictions.map((juris) => ({
+            id: juris.id,
+          })),
+        },
+        userRoles:
+          'userRoles' in dto
+            ? {
+                create: {
+                  ...dto.userRoles,
+                },
+              }
+            : undefined,
+        listings: dto.listings
+          ? {
+              connect: dto.listings.map((listing) => ({
+                id: listing.id,
+              })),
+            }
+          : undefined,
+      },
+    });
+
+    const confirmationToken = this.createConfirmationToken(
+      newUser.id,
+      newUser.email,
+    );
+    newUser = await this.prisma.userAccounts.update({
+      include: view,
+      data: {
+        confirmationToken: confirmationToken,
+      },
+      where: {
+        id: newUser.id,
+      },
+    });
+
+    if (!forPartners && sendWelcomeEmail) {
+      const confirmationUrl = this.getPublicConfirmationUrl(
+        dto.appUrl,
+        confirmationToken,
+      );
+      // TODO: email service
+    } else if (
+      forPartners &&
+      existingUser &&
+      'userRoles' in dto &&
+      existingUser?.userRoles?.isPartner &&
+      dto?.userRoles?.isPartner &&
+      this.jurisdictionMismatch(dto.jurisdictions, existingUser.jurisdictions)
```

Comment:
What is this use case?

Reply:
Sorry, one additional question, do we also need this to cover the case of the existing user is a "jurisdictional Admin" and not just a partner?

Reply:
that's a good question, right now a jurisdictional admin can only be tied to one jurisdiction (or rather the front end only supports them being tied to one jurisdiction) 

So my guts tell me not to support that case right now, and address it when we look at giving jurisdictional admin access to several jurisdictions if that makes sense